### ISSUE TITLE
no-ignored-return required usage

### ISSUE THREAD

**Issue by mfbx9da4** [SonarSource/eslint-plugin-sonarjs#315]
Enforce for a function no-ignored-return

**Reproducer**

```javascript
function hello() {
  return 'hi'.lastIndexOf('a')
}

hello()
```

**Expected behavior**

it should throw error 

```
The return value of "hello" must be used.eslintsonarjs/no-ignored-return
```

**I want to provide the following feedback.**

Ideally it should infer the return value is not used. Alternatively, is there a way to specify `hello()` return value should be used?

**Example**

**eslint-plugin-sonarjs version:** 0.11.0<br>
**eslint version:** 7.14.0<br>
**Node.js version:** 14.17.0<br>

**Rule key:**no-ignored-return

**Comment by vilchik-elena** [comment]
Hi,

This rule only relies on built-in functions where we are sure there is no side effect, other than returned value. We hardly can be sure about it for user-defined functions. 
Also we don't think that making it configurable is a good idea, we can't expect user to maintain it properly also there is no proper naming for functions in JS.
